system-addon/content-src/components/TopSites/TopSiteForm.jsx

```diff
@@ -108,32 +199,40 @@ export class TopSiteForm extends React.PureComponent {
                 placeholderId="topsites_form_title_placeholder"
                 intl={this.props.intl} />
               <TopSiteFormInput onChange={this.onUrlChange}
+                shouldFocus={this.state.validationError && !this.validateUrl(this.state.url)}
                 value={this.state.url}
                 onClear={this.onClearUrlClick}
-                validationError={this.state.validationError}
+                validationError={this.state.validationError && !this.validateUrl(this.state.url)}
                 titleId="topsites_form_url_label"
                 typeUrl={true}
                 placeholderId="topsites_form_url_placeholder"
                 errorMessageId="topsites_form_url_validation"
                 intl={this.props.intl} />
+              {this._renderCustomScreenshotInput()}
             </div>
-            <TopSiteLink link={this.props.site || {}} title={this.state.label} />
+            <TopSiteLink link={this._getPreviewScreenshot()}
+              defaultStyle={this.props.screenshotRequestFailed}
```

Comment:
Did design confirm that a failed preview should show the letter fallback vs revert to what it showed before the preview attempt?

Reply:
I showed Aaron a screenshot of a failed request (red message, letter fallback), and he said "perfect. seems clear to me" So this should be good.